dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataelement/DefaultDataElementService.java

```diff
@@ -80,18 +70,39 @@ public DefaultDataElementService( DataElementStore dataElementStore,
     @Transactional
     public long addDataElement( DataElement dataElement )
     {
+        validateDateElement( dataElement );
         dataElementStore.save( dataElement );
-
         return dataElement.getId();
     }
 
     @Override
     @Transactional
     public void updateDataElement( DataElement dataElement )
     {
+        validateDateElement( dataElement );
         dataElementStore.update( dataElement );
     }
 
+    @Override
+    public void validateDateElement( DataElement dataElement )
+    {
+        if ( dataElement.getOptionSet() == null )
+        {
+            if ( dataElement.getValueType() == ValueType.MULTI_TEXT )
+            {
+                throw new IllegalQueryException( new ErrorMessage( ErrorCode.E1116, dataElement.getUid() ) );
+            }
+            return;
+        }
+        // need to reload the options in case this is a create and the options
+        // is a shallow reference object
+        OptionSet options = optionSetStore.getByUid( dataElement.getOptionSet().getUid() );
```

Comment:
Seems the comment above doesn't match with the code here. Maybe we don't need to reload OptionSet.

Reply:
In what why doesn't it match? the `OptionSet` is reloaded directly after the comment as it states. It was needed because otherwise the set might be a reference what does not hold the value type information. This occurs when a new data element with option set reference is created as the comment says. I assume this also would occur when an existing data element gets assigned a option set.

Reply:
ok then its all good. 
Was a bit confused with the variable name `OptionSet options = ...`